lib/robots/dor_repo/ocr/fetch_files.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Robots
+  module DorRepo
+    module Ocr
+      # Fetch files in need of OCR from Preservation
+      class FetchFiles < LyberCore::Robot
+        def initialize
+          super('ocrWF', 'ocr-fetch-files')
+        end
+
+        def perform_work
+          copy_files
+        end
+
+        private
+
+        def copy_files
+          ocrable_filenames.each do |filename|
+            File.open(abbyy_path(filename), 'wb') do |file_writer|
+              preservation_client.objects.content(druid:,
+                                                  filepath: filename,
+                                                  on_data: proc { |data, _count| file_writer.write(data) })
+            end
+          end
+        end
+
+        def abbyy_path(filename)
+          # NOTE: if files of type "file" were allowed here we would have to
+          # deal with file hierarchy (subdirectories)
+          File.join(Settings.sdr.abbyy_ticket_path, filename)
+        end
```

Comment:
We are going to organize the files by druid, so that the folder structure will be something like:

`/abbyy/INPUT/ab123df4567/files...`

So I think this should be something like:

```
 File.join(Settings.sdr.abbyy_ticket_path, druid, filename)
```

And I think we need to create that druid subfolder if it doesn't exist.  So something like this at the start of `#copy_files`:

```
FileUtils.mkdir(File.join(Settings.sdr.abbyy_ticket_path, druid)) unless unless File.directory?(File.join(Settings.sdr.abbyy_ticket_path, druid))
```

But I guess to avoid all the code duplication, could add this private method instead and then replace all the duplicated code above with the method:

```
def base_abbyy_folder
  File.join(Settings.sdr.abbyy_ticket_path, druid)
end
```

Reply:
Which reminds me, we probably need a new "cleanup abbyy folder" robot at the end to clear out the druid folder.  Even though ABBYY clears out the files themselves after OCRing, i think the empty folder will remain.  I'll ticket that.

Reply:
> We are going to organize the files by druid, so that the folder structure will be something like:
> 
> `/abbyy/INPUT/ab123df4567/files...`

It looks like the ticket is getting written to `/abbyy/INPUT/` https://github.com/sul-dlss/common-accessioning/blob/main/lib/dor/text_extraction/abbyy.rb#L21-L23

And the file paths in the ticket reference the files as being alongside the ticket: https://github.com/sul-dlss/common-accessioning/blob/main/lib/dor/text_extraction/abbyy.rb#L64-L66

Should the ticket still get written to that location but reference the file in the druid subdirectory?

Reply:
I think the way this XML writing code is currently written, it is assuming you will pass in an array of filenames where the filepaths include the "druid" subfolder each time.  This is the most flexible but perhaps not ideal.  Maybe we need to put this kind of logic (i.e. the "base folder" for the files) in one place (e.g. in the `Ocr` class) and then use it in both places?

Reply:
I've modified how this works over in #1215 so that the caller just knows the filenames (possible hierarchical) and the Abbyy class takes care of naming things with DOS path separator, etc.

Reply:
The code should be updated now to write to a druid subdirectory.